Add groups pruning section

soltysh opened the pull request:
@openshift/team-documentation ptal

Additionally, where can I put some release notes I have two topics for them:
1. this one
2. another about deprecating an env variable used by `oc`.

soltysh commented:
@ahardin-rh || @vikram-redhat ptal

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+.^|`--blacklist`
+|Path to the group blacklist file.
+See xref:../install_config/syncing_groups_with_ldap.adoc#overview[Syncing Groups With LDAP]
+for the structure of this file.
```
s/structure of this file./blacklist file structure.

I don't actually see the blacklist or whitelist file structure in this link.

soltysh replied:
I know :( once it'll get added there the link will be ok :)

kalexand-rh replied:
Right, but until then, it's wrong. Do you have the format? If so, we should add it before this change is merged.

soltysh replied:
I don't, I'll try to find one.

soltysh replied:
Let's have it as a followup to this one.

kalexand-rh replied:
Will you check? If we don't add the whitelist/blacklist format to the other topic, then we need to modify this one to not claim that the other topic has it.

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+.^|`--whitelist`
+|Path to the group whitelist file.
+See xref:../install_config/syncing_groups_with_ldap.adoc#overview[Syncing Groups With LDAP]
+for the structure of this file.
```
s/structure of this file./whitelist file structure.

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+
+== Pruning groups
+
+In order to prune groups records from an external provider, administrators may
```
Will you clarify what kind of provider? Is it just LDAP? Are these just LDAP groups?

soltysh replied:
Yeah, currently it's just LDAP, but I'd rather keep it more general so that I don't need to update it in the future ;)

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+
+.^|`--sync-config`
+|Path to the synchronization configuration file.
+See xref:../install_config/syncing_groups_with_ldap.adoc#overview[Syncing Groups With LDAP]
```
I'd link to that section: ../install_config/syncing_groups_with_ldap.html#configuring-ldap-sync[Configuring LDAP Sync]

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+for the structure of this file.
+|===
+
+To see what a pruning operation would delete:
```
s/what a pruning operation would delete:/the groups that the prune command deletes:

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+$ oc adm prune groups --sync-file=ldap-sync-config.yaml
+---
+
+To actually perform the prune operation:
```
s/actually perform/perform

kalexand-rh commented:
@stuartchuan, will you PTAL?

kalexand-rh commented:
Also, will one of you confirm that this change is valid back to 3.5?

stuartchuan commented:
@kalexand-rh， seems this is redundant to  https://docs.openshift.com/container-platform/3.9/install_config/syncing_groups_with_ldap.html#running-a-group-pruning-job

soltysh commented:
> @kalexand-rh， seems this is redundant to https://docs.openshift.com/container-platform/3.9/install_config/syncing_groups_with_ldap.html#running-a-group-pruning-job

I've change that section slightly and linked it back to this new one.

> Also, will one of you confirm that this change is valid back to 3.5?

No, this is 3.10, only.

@kalexand-rh @stuartchuan I've addressed all the comments, ptal

kalexand-rh commented on admin_guide/pruning_resources.adoc:
```diff
@@ -33,10 +33,58 @@ $ oc adm prune <object_type> <options>
+for the structure of this file.
+|===
+
+To see the groups that the prune command deletes::
```
There's an extra ":" here.

soltysh replied:
Fixed.

kalexand-rh commented:
@soltysh, it's looking pretty good. I have a question and found a typo. :)

stuartchuan commented:
@soltysh Thanks for the update, others lgtm.

kalexand-rh commented:
@soltysh, do you have any news on that whitelist/blacklist format?

soltysh commented:
> @soltysh, do you have any news on that whitelist/blacklist format?

There's this information in that link I've pointed to `Your files must contain one unique group identifier per line.`, after I've talked with folks who wrote it, I think that's sufficient :)

kalexand-rh commented:
@soltysh, that sounds fair. I might open a follow-up PR to clarify that note block a little. Thanks!

kalexand-rh commented:
/cherrypick enterprise-3.10
